templates/opat-0.1.0.handlebars

```diff
@@ -0,0 +1,50 @@
+# {{title}}
+
+Based on {{catalog.title}}
+
+## Name of Product/Version
+{{product.name}}{{#if product.version}} {{product.version}}{{/if}}
+
+## Report Date
+{{ now }}
+
+{{#if product.descriptipn}}
+## Product Description
+{{product.description}}
+{{/if}}
+
+## Contact Information
+{{#if contact.name}}Name: {{contact.name}}{{/if}}
+{{#if contact.address}}Address: {{contact.address}}{{/if}}
+Email: {{contact.email}}
+{{#if contact.phone}}Phone: {{contact.phone}}{{/if}}
+
+## Notes
+{{notes}}
+
+## Terms
+The terms used in the Conformance Level information are defined as follows:
+- **Supports**: The functionality of the product has at least one method that meets the criterion without known defects or meets with equivalent facilitation.
+- **Partially Supports**: Some functionality of the product does not meet the criterion.
+- **Does Not Support**: The majority of product functionality does not meet the criterion.
+- **Not Applicable**: The criterion is not relevant to the product.
+- **Not Evaluated**: The product has not been evaluated against the criterion. This can be used only in WCAG 2.0 Level AAA.
+
+{{#each catalog.chapters as |catalog-chapter|}}
```

Comment:
I like that we can just treat chapters as a loop in the basic template, since that should work find in the general case and vastly reduces the amount of code. I assume folks could still break these out and add the markup bit by bit if they wanted to customize the layout of specific chapters, right?

Reply:
Agreed. I can see if I can refactor it.

Reply:
Actually now I remembered. Because in the OPAT example, chapters just have special keys it was easier to loop using the catalog. Maybe I am misreading your comment.

Reply:
@dmundra I think being able to loop is ideal, but it would be nice if you could also choose to format some or all of the chapters separately (for example, perhaps you want to include a column linking to EARL results in the WCAG sections but not in the others).
I am not very familiar with handlebars, but this might look like accessing `catalog.chapters.success_criteria_level_a` or something like that - it wasn't obvious if that was possible with the current structure or not. If switching to an object would make it impossible/inconvenient to loop, perhaps we can provide separate `chaptersList` and `chapters` inputs to the template?

Reply:
Ah yes, formating the chapters separately is definitely doable and I would switch the handlebars code to start with the data from the OPAT instead of catalog and grab values from the catalog as needed. The code could look something like this and would need a helper (as your example wouldn't work but 'helpers' can bridge the gap):


```typescript
    Handlebars.registerHelper(
      "catalogChapter",
      function (chapterId) {
        for (const chapter of catalogData.chapters) {
          if (chapter.id === chapterId) {
            return chapter;
          }
        }
      }
    );
```

```handlebars
{{#with (catalogChapter 'success_criteria_level_a') as |catalog-chapter|}}
## {{chapterOrTableTitle catalog-chapter}} {{catalog-chapter.order}}: {{catalog-chapter.label}}
{{#if chapters.success_criteria_level_a.notes}}
Notes: {{chapters.success_criteria_level_a.notes}}
{{/if}}
{{/with}}
```

Reply:
I think we should kick the can for customizations down the road @grugnog. I updated #15 with a small note about that.